// @aws-sdk/client-ecs/dist-es/commands/DescribeCapacityProvidersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeCapacityProvidersRequest, DescribeCapacityProvidersResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeCapacityProvidersCommand, serializeAws_json1_1DescribeCapacityProvidersCommand, } from "../protocols/Aws_json1_1";
var DescribeCapacityProvidersCommand = (function (_super) {
    __extends(DescribeCapacityProvidersCommand, _super);
    function DescribeCapacityProvidersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCapacityProvidersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "DescribeCapacityProvidersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCapacityProvidersRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeCapacityProvidersResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCapacityProvidersCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeCapacityProvidersCommand(input, context);
    };
    DescribeCapacityProvidersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeCapacityProvidersCommand(output, context);
    };
    return DescribeCapacityProvidersCommand;
}($Command));
export { DescribeCapacityProvidersCommand };
