// @aws-sdk/client-ecs/dist-es/commands/DescribeTaskDefinitionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeTaskDefinitionRequest, DescribeTaskDefinitionResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeTaskDefinitionCommand, serializeAws_json1_1DescribeTaskDefinitionCommand, } from "../protocols/Aws_json1_1";
var DescribeTaskDefinitionCommand = (function (_super) {
    __extends(DescribeTaskDefinitionCommand, _super);
    function DescribeTaskDefinitionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeTaskDefinitionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "DescribeTaskDefinitionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeTaskDefinitionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeTaskDefinitionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeTaskDefinitionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeTaskDefinitionCommand(input, context);
    };
    DescribeTaskDefinitionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeTaskDefinitionCommand(output, context);
    };
    return DescribeTaskDefinitionCommand;
}($Command));
export { DescribeTaskDefinitionCommand };
