// @aws-sdk/client-ecs/dist-es/commands/ListTasksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTasksRequest, ListTasksResponse } from "../models/models_0";
import { deserializeAws_json1_1ListTasksCommand, serializeAws_json1_1ListTasksCommand } from "../protocols/Aws_json1_1";
var ListTasksCommand = (function (_super) {
    __extends(ListTasksCommand, _super);
    function ListTasksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTasksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "ListTasksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTasksRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListTasksResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTasksCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListTasksCommand(input, context);
    };
    ListTasksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListTasksCommand(output, context);
    };
    return ListTasksCommand;
}($Command));
export { ListTasksCommand };
