// @aws-sdk/client-ecs/dist-es/commands/ListAttributesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListAttributesRequest, ListAttributesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListAttributesCommand, serializeAws_json1_1ListAttributesCommand, } from "../protocols/Aws_json1_1";
var ListAttributesCommand = (function (_super) {
    __extends(ListAttributesCommand, _super);
    function ListAttributesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "ListAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListAttributesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListAttributesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListAttributesCommand(input, context);
    };
    ListAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListAttributesCommand(output, context);
    };
    return ListAttributesCommand;
}($Command));
export { ListAttributesCommand };
