// @aws-sdk/client-ecs/dist-es/commands/RegisterContainerInstanceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterContainerInstanceRequest, RegisterContainerInstanceResponse } from "../models/models_0";
import { deserializeAws_json1_1RegisterContainerInstanceCommand, serializeAws_json1_1RegisterContainerInstanceCommand, } from "../protocols/Aws_json1_1";
var RegisterContainerInstanceCommand = (function (_super) {
    __extends(RegisterContainerInstanceCommand, _super);
    function RegisterContainerInstanceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RegisterContainerInstanceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "RegisterContainerInstanceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RegisterContainerInstanceRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RegisterContainerInstanceResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RegisterContainerInstanceCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1RegisterContainerInstanceCommand(input, context);
    };
    RegisterContainerInstanceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1RegisterContainerInstanceCommand(output, context);
    };
    return RegisterContainerInstanceCommand;
}($Command));
export { RegisterContainerInstanceCommand };
