// @aws-sdk/client-ecs/dist-es/commands/UpdateClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateClusterRequest, UpdateClusterResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateClusterCommand, serializeAws_json1_1UpdateClusterCommand, } from "../protocols/Aws_json1_1";
var UpdateClusterCommand = (function (_super) {
    __extends(UpdateClusterCommand, _super);
    function UpdateClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "UpdateClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateClusterRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateClusterResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateClusterCommand(input, context);
    };
    UpdateClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateClusterCommand(output, context);
    };
    return UpdateClusterCommand;
}($Command));
export { UpdateClusterCommand };
