// @aws-sdk/client-ecs/dist-es/commands/UpdateContainerInstancesStateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateContainerInstancesStateRequest, UpdateContainerInstancesStateResponse } from "../models/models_0";
import { deserializeAws_json1_1UpdateContainerInstancesStateCommand, serializeAws_json1_1UpdateContainerInstancesStateCommand, } from "../protocols/Aws_json1_1";
var UpdateContainerInstancesStateCommand = (function (_super) {
    __extends(UpdateContainerInstancesStateCommand, _super);
    function UpdateContainerInstancesStateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateContainerInstancesStateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ECSClient";
        var commandName = "UpdateContainerInstancesStateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateContainerInstancesStateRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateContainerInstancesStateResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateContainerInstancesStateCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateContainerInstancesStateCommand(input, context);
    };
    UpdateContainerInstancesStateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateContainerInstancesStateCommand(output, context);
    };
    return UpdateContainerInstancesStateCommand;
}($Command));
export { UpdateContainerInstancesStateCommand };
